fcrepo-kernel-api/src/main/java/org/fcrepo/kernel/api/identifiers/FedoraID.java

```diff
@@ -156,6 +180,56 @@ public String getMementoString() {
         return mementoDatetimeStr;
     }
 
+    /**
+     * Add a part onto this identifier.
+     * @param addition The additional string.
+     * @return new FedoraID for the new identifier.
+     */
+    public FedoraID addToResourceId(final String addition) {
+        return FedoraID.create(this.getResourceId() + (this.getResourceId().endsWith("/") ? "" : "/") + addition);
+    }
+
+    /**
+     * Add a part onto this identifier.
+     * @param addition The additional string.
+     * @return new FedoraID for the new identifier.
+     */
+    public FedoraID addToFullId(final String addition) {
+        return FedoraID.create(this.getFullId() + (this.getFullId().endsWith("/") ? "" : "/") + addition);
+    }
+
+    @Override
+    public boolean equals(final Object obj) {
+        if (obj == this) {
+            return true;
+        }
+
+        if (!(obj instanceof FedoraID)) {
+            return false;
+        }
+
+        final var testObj = (FedoraID) obj;
+        return testObj.getFullId().equals(this.getFullId());
```

Comment:
Generally, using `Objects.equals()` is safer. In this case, it's probably impossible for `fullId` to be null?

Reply:
It is impossible, but I can change it.